### PR TITLE
ROC-3059: Payment api v2 related changes

### PR DESCRIPTION
**Pull request by singh-surjeet**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-3059

### Change description ###
Description field and associated usage

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/assertion/CCDPaymentAssert.java [hunk 1]
```diff
@@ -22,7 +22,7 @@ public CCDPaymentAssert isEqualTo(Payment payment) {
 
         if (!Objects.equals(actual.getStatus(), payment.getStatus())) {
             failWithMessage("Expected CCDPayment.status to be <%s> but was <%s>",
-                payment.getStatus(), actual.getDescription());
+                payment.getStatus());
```
Comment (sabahirfan) [inline ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/assertion/CCDPaymentAssert.java]: 2nd placeholder is missing, it should be `actual.getStatus()`

File: ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/assertion/PaymentAssert.java [hunk 2]
```diff
@@ -22,7 +22,7 @@ public PaymentAssert isEqualTo(CCDPayment payment) {
 
         if (!Objects.equals(actual.getStatus(), payment.getStatus())) {
             failWithMessage("Expected Payment.status to be <%s> but was <%s>",
-                payment.getDescription(), actual.getStatus());
+                actual.getStatus());
```
Comment (sabahirfan) [inline ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/assertion/PaymentAssert.java]: 2nd placeholder is missing, it should be payment.getStatus()

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CCDPayment.java [hunk 3]
```diff
@@ -11,7 +11,6 @@
     private String id;
     private BigDecimal amount;
     private String reference;
-    private String description;
```
Comment (sabahirfan) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CCDPayment.java]: Removing description field from ccd adapter does not look correct?

Comment (sabahirfan) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CCDPayment.java]: It's fine as we are not using it.

Comment (singh-surjeet) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CCDPayment.java]: All actioned!

**Pull request merged**